### PR TITLE
Add SearchResult from proto convesion 

### PR DESCRIPTION
**Pull request by vthacker**
Add SearchResult from proto convesion 

We need this for aggregation in the query service

### CODE REVIEW

File: src/main/java/com/slack/kaldb/server/KaldbLocalSearcher.java [hunk 1]
```diff
@@ -42,6 +65,7 @@ public static SearchQuery fromSearchRequest(KaldbSearch.SearchRequest searchRequ
     searchResultBuilder.setFailedNodes(searchResult.failedNodes);
     searchResultBuilder.setTotalNodes(searchResult.totalNodes);
     searchResultBuilder.setTotalSnapshots(searchResult.totalSnapshots);
+    searchResultBuilder.setSnapshotsWithReplicas(searchResult.snapshotsWithReplicas);
```
Comment (vthacker) [inline src/main/java/com/slack/kaldb/server/KaldbLocalSearcher.java:68]: was missing

File: src/main/java/com/slack/kaldb/logstore/LogMessage.java [hunk 2]
```diff
@@ -142,4 +142,17 @@ public void addProperty(String key, Object value) {
   public LogWireMessage toWireMessage() {
     return new LogWireMessage(getIndex(), getType(), id, source);
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    LogMessage that = (LogMessage) o;
+    return timeSinceEpochMilli == that.timeSinceEpochMilli;
```
Comment (bryanlb) [inline src/main/java/com/slack/kaldb/logstore/LogMessage.java:151]: This doesn't seem right - we would expect to see multiple messages with identical timestamps.

**Review by bryanlb — CHANGES_REQUESTED** [review]

File: src/main/java/com/slack/kaldb/logstore/LogMessage.java (hunk 2 above)
Comment (vthacker) [reply src/main/java/com/slack/kaldb/logstore/LogMessage.java:151]: it does not! Unsure what else fields can I use to compare?

File: src/test/java/com/slack/kaldb/server/SearchResultTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    assertThat(protoSearchResult.getFailedNodes()).isEqualTo(1);
+    assertThat(protoSearchResult.getTotalNodes()).isEqualTo(5);
+    assertThat(protoSearchResult.getTotalSnapshots()).isEqualTo(7);
+    assertThat(protoSearchResult.getSnapshotsWithReplicas()).isEqualTo(7);
+    assertThat(protoSearchResult.getBucketsCount()).isEqualTo(1);
+
+    SearchResult<LogMessage> convertedSearchResult =
+        KaldbLocalSearcher.fromSearchResultProto(protoSearchResult);
+
+    assertThat(convertedSearchResult).isEqualTo(searchResult);
```
Comment (mansu) [inline src/test/java/com/slack/kaldb/server/SearchResultTest.java:52]: Can you please an additional test using a hashmap or a set to test the hashing logic?

File: src/main/java/com/slack/kaldb/logstore/LogMessage.java (hunk 2 above)
Comment (bryanlb) [reply src/main/java/com/slack/kaldb/logstore/LogMessage.java]: Per our discussion, `_source` will be expected to contain a significant amount of unique attributes. We probably should ensure this map is the same across objects.

**Review by bryanlb — DISMISSED** [review]

**Review by bryanlb — APPROVED** [review]

**Pull request merged**